src/ng-select/lib/ng-select.component.ts

```diff
@@ -604,50 +602,54 @@ export class NgSelectComponent implements OnDestroy, OnChanges, OnInit, AfterVie
 
 	selectTag() {
 		let tag;
-		if (isFunction(this.addTag)) {
-			tag = (<AddTagFn>this.addTag)(this.searchTerm);
+		const addTag = this.addTag();
+		if (isFunction(addTag)) {
+			tag = (<AddTagFn>addTag)(this.searchTerm);
 		} else {
-			tag = this._primitive ? this.searchTerm : { [this.bindLabel]: this.searchTerm };
+			tag = this._primitive ? this.searchTerm : { [this.bindLabel()]: this.searchTerm };
 		}
 
-		const handleTag = (item) => (this._isTypeahead || !this.isOpen ? this.itemsList.mapItem(item, null) : this.itemsList.addItem(item));
+		const handleTag = (item) => (this._isTypeahead || !this.isOpen() ? this.itemsList.mapItem(item, null) : this.itemsList.addItem(item));
 		if (isPromise(tag)) {
-			tag.then((item) => this.select(handleTag(item))).catch(() => {});
+			tag.then((item) => this.select(handleTag(item))).catch(() => { });
 		} else if (tag) {
 			this.select(handleTag(tag));
 		}
 	}
 
 	showClear() {
-		return this.clearable && (this.hasValue || this.searchTerm) && !this.disabled;
+		return this.clearable() && (this.hasValue || this.searchTerm) && !this.disabled;
 	}
 
 	focusOnClear() {
 		this.blur();
-		if (this.clearButton) {
-			this.clearButton.nativeElement.focus();
+		const clearButton = this.clearButton();
+		if (clearButton) {
+			clearButton.nativeElement.focus();
 		}
 	}
 
 	trackByOption = (_: number, item: NgOption) => {
-		if (this.trackByFn) {
-			return this.trackByFn(item.value);
+		const trackByFn = this.trackByFn();
```

Comment:
Nit: I don't we need separate variable here

Reply:
Updated - https://github.com/ng-select/ng-select/pull/2584/commits/53cf59e5182d0cbc93ac9608a82ee5b0dc0c44be